src/libs/richJson/richJson.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ *
+ * richJson lib
+ *
+ * JSON.serialize and JSON.parse don't support BigInt values.
+ * To address this limitation, we have created this small library that adds support for BigInt numbers
+ * during JSON serialization and parsing.
+ *
+ * Limitations: The library does not currently support BigInt values in new Map, Set, or Uint8Array.
+ * However, extending and adding support can be easily accomplished if needed.
+ * @credits: https://dev.to/benlesh/bigint-and-json-stringify-json-parse-2m8p
+ *
+ *
+ * Additionally, JSON.serialize and JSON.parse do not properly serialize the Error object, so we extend that functionality here as well.
+ */
+
+export function stringify(obj: any): string {
+  return JSON.stringify(obj, (key, value) => {
+    if (typeof value === 'bigint') {
+      return { $bigint: value.toString() }
+    }
+
+    if (value instanceof Error) {
+      const error: any = {}
```

Comment:
won't this work with spread, Object.assign or something

Reply:
tried spreading and Object.assign but it doesn't work and after stringifying the result is an empty object.

only 
`    
  const error: any = {}
  Object.getOwnPropertyNames(value).forEach((propName) => {
     error[propName] = value[propName]
  })
  return error
`
and
`
  return Object.getOwnPropertyNames(value).reduce((error: any, propName: string) => {
    error[propName] = value[propName];
    return error;
  }, {});
`
seem to do the trick